### ISSUE TITLE
Fix accessibility issues from Pa11y audit: duplicate ids

### ISSUE THREAD

**Issue by brentryanjohnson** [ONRR/doi-extractives-data#4068]
• Error: Duplicate id attribute value "backgroundRect" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #backgroundRect
└──

• Error: Duplicate id attribute value "maxExtent" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #maxExtent
└── <text width="300" x="300" y="15...

• Error: Duplicate id attribute value "bars" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #bars
└── <g height="175" width="21.21212...

• Error: Duplicate id attribute value "groups" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #groups
└── <line x1="4.242424242424243" x2...

• Error: Duplicate id attribute value "backgroundRect" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #backgroundRect
└──

• Error: Duplicate id attribute value "maxExtent" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #maxExtent
└── <text width="300" x="300" y="15...

• Error: Duplicate id attribute value "bars" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #bars
└── <g height="175" width="21.21212...

• Error: Duplicate id attribute value "groups" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #groups
└── <line x1="4.242424242424243" x2...

• Error: Duplicate id attribute value "backgroundRect" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #backgroundRect
└──

• Error: Duplicate id attribute value "maxExtent" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #maxExtent
└── <text width="300" x="300" y="15...

• Error: Duplicate id attribute value "bars" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #bars
└── <g height="175" width="21.21212...

• Error: Duplicate id attribute value "groups" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #groups
└── <line x1="4.242424242424243" x2...

• Error: Duplicate id attribute value "backgroundRect" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #backgroundRect
└──

• Error: Duplicate id attribute value "maxExtent" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #maxExtent
└── <text width="349" x="349" y="15...

• Error: Duplicate id attribute value "bars" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #bars
└── <g height="175" width="24.67676...

• Error: Duplicate id attribute value "groups" found on the web page.
├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
├── #groups
└── <line x1="4.935353535353537" x2...

**Comment by jlouis-onrr** [comment]
@brentryanjohnson What pa11y command produced that output?

**Comment by brentryanjohnson** [comment]
@jlouis-onrr I think a lot of these might have been resolved. I think we need a new issue with a new audit output (this one is from May).

But the [pa11y documentation is here](https://github.com/pa11y/pa11y).

This was produced with the basic command: `pa11y https://revenuedata.doi.gov`.

**Comment by jlouis-onrr** [comment]
Great to know, i was seeing the following with the same command, seems the errors have changed since the start of the thread.

Running Pa11y on URL https://revenuedata.doi.gov

Results for URL: https://revenuedata.doi.gov/

 • Error: Heading tag found with no content. Text that is not intended as a heading should not be marked up with heading tags.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.H42.2
   ├── #main-content > h3
   └── <h3 class="h3-bar"></h3>

 • Error: This select element does not have a name available to an accessibility API. Valid names are: label element, title undefined, aria-label undefined, aria-labelledby undefined.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Select.Name
   ├── #gatsby-focus-wrapper > div > main > div > div > div:nth-child(2) > section > div:nth-child(2) > section > div > div:nth-child(1) > div:nth-child(2) > div > select
   └── <select><option value="calendar_year">C...</select>

 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrapped around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #gatsby-focus-wrapper > div > main > div > div > div:nth-child(2) > section > div:nth-child(2) > section > div > div:nth-child(1) > div:nth-child(2) > div > select
   └── <select><option value="calendar_year">C...</select>

 • Error: Duplicate id attribute value "Layer_1" found on the web page.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
   ├── #Layer_1
   └── <svg version="1.1" id="Layer_1" x="0px" y="0px" viewBox="0 0 100 100" width="25px" height="25px" xml:space="preserve"><style type="text/css"> .st0{f...</svg>

4 Errors

**Comment by jlouis-onrr** [comment]
Damn... look what 18F was up to, getting Pa11y to run or crawl all pages... just what I needed!

Get pa11y to crawl all pages
[pa11y-crawl](https://github.com/18F/pa11y-crawl)

**Comment by jlouis-onrr** [comment]
:sob: no worky... spewing errors, guess I will go page by page

**Comment by jlouis-onrr** [comment]
adding errors found on Explore Data:

$ pa11y --wait 1000 --debug http://localhost:8000/explore/

Results for URL: http://localhost:8000/explore/

 • Error: Duplicate id attribute value "production" found on the web page.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
   ├── #production
   └── <h2 class="state-page-overview" id="production">Production</h2>

 • Error: This select element does not have a name available to an accessibility API. Valid names are: label element, title undefined, aria-label undefined, aria-labelledby undefined.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Select.Name
   ├── #federal-production-trends-by-resources-2018-2017-2016-2015-2014-2013-2012-2011-2010-2009 > select
   └── <select class="chart-selector flex-row-icon"><option value="2018">2018</opti...</select>

 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrapped around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #federal-production-trends-by-resources-2018-2017-2016-2015-2014-2013-2012-2011-2010-2009 > select
   └── <select class="chart-selector flex-row-icon"><option value="2018">2018</opti...</select>

 • Error: Duplicate id attribute value "revenue" found on the web page.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
   ├── #revenue
   └── <h2 id="revenue">Revenue</h2>

 • Error: Not all th elements in this table have a scope attribute. These cells should contain a scope attribute to identify their association with td elements.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.H63.1
   ├── #revenue-types
   └── <table id="revenue-types" is="bar-chart-table" class="revenue table-arrow_box"><thead id="revenue-phase"><tr><...</table>

 • Error: Duplicate id attribute value "icon-coal" found on the web page.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
   ├── #icon-coal
   └── <g id="icon-coal"><g id="coal-icon-resources" dat...</g>

 • Error: Duplicate id attribute value "coal-icon-resources" found on the web page.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
   ├── #coal-icon-resources
   └── <g id="coal-icon-resources" data-name="1.0-Resources-to-Revenue-Landing"><g><g><path d="M24.05,40.18a4.3...</g>

 • Error: This select element does not have a name available to an accessibility API. Valid names are: label element, title undefined, aria-label undefined, aria-labelledby undefined.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Select.Name
   ├── #revenue-trends > select
   └── <select class="chart-selector flex-row-icon"><option value="2018">2018</opti...</select>

 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrapped around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #revenue-trends > select
   └── <select class="chart-selector flex-row-icon"><option value="2018">2018</opti...</select>

 • Error: This select element does not have a name available to an accessibility API. Valid names are: label element, title undefined, aria-label undefined, aria-labelledby undefined.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Select.Name
   ├── #by-fund > select
   └── <select class="chart-selector flex-row-icon"><option value="2018">2018</opti...</select>

 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrapped around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #by-fund > select
   └── <select class="chart-selector flex-row-icon"><option value="2018">2018</opti...</select>

11 Errors

**Comment by jlouis-onrr** [comment]
Maybe we should open new issues for any additional pages that have accessibility issues? Since this issue seems to only address the homepage.
